src/server/lib/gameState.ts

```
@@ -437,9 +466,24 @@ const drawCards = (playerIDs: string[]) => {
     fullDeck.length / playerIDs.length
   );
   for (let i = 0; i < playerIDs.length; i++) {
-    // Take 5 letters from the chunk and assign it to the player
-    const word = chunks[i].splice(0, 5);
-    playerHands[playerIDs[i]] = word;
+    // We really just need one word; getting anagrams gets all valid
+    // word permutations and can be costly for long lists of characters
+    // Cut down the size of the chunk
+    let possibleWords = [];
+    for (let j = 5; j < chunks[i].length; j++) {
+      possibleWords = wordTree.getSubAnagrams(chunks[i].slice(j - 5, j).join());
+      if (possibleWords.length !== 0) {
+        break;
+      }
+    }
+    if (possibleWords.length === 0) {
+      console.error(`No words found for player ${i}`);
```

Maybe return an error and re-run `drawCards` in a loop until it can initialize with a valid game state.

---

I see you commented about this case in the Trello card 👍

---

Added some reshuffling logic here: 7cd32c4